agent/command/git.go

```diff
@@ -236,6 +247,18 @@ func (opts cloneOpts) buildHTTPCloneCommand(forApp bool) ([]string, error) {
 	if opts.branch != "" {
 		clone = fmt.Sprintf("%s --branch '%s'", clone, opts.branch)
 	}
+	// If we're not falling back to a full clone, we can use single-branch and filter.
+	if !opts.fallbackToFullClone {
+		if !opts.cloneAllBranches {
```

Comment:
Do you think there's user cases to enable cloning all branches but not full tree (or vice versa)? Wondering if there was a rationale for splitting it into two options or if it just seemed like a nice user experience (which it is! so no change needed, just curious)

Reply:
Since both apply a different field, it keeps the precedence of the other options and it helps with clarity- if they were combined it would either have to be a big name to explain both or it would have to just be an abstract name like `FasterClone` which they would need Evergreen specific context to interpret.